Library/Homebrew/readall.rb

```diff
@@ -24,16 +24,15 @@ def valid_ruby_syntax?(ruby_files)
       !failed
     end
 
-    def valid_aliases?(alias_dirs)
+    def valid_aliases?(alias_dir, formula_dir)
+      return false unless alias_dir.directory?
+
       failed = false
-      alias_dirs.each do |alias_dir|
-        next unless alias_dir.directory?
-        alias_dir.children.each do |f|
-          next unless f.symlink?
-          next if f.file?
-          onoe "Broken alias: #{f}"
-          failed = true
-        end
+      alias_dir.each_child do |f|
+        next unless f.symlink?
+        next if f.file? && !(formula_dir/"#{f.basename}.rb").exist?
+        onoe "Broken alias: #{f}"
+        failed = true
       end
       !failed
```

Comment:
This might be better with an `any?`:

```ruby
alias_dir.each_child.any? do |f|
  next unless f.symlink?
  next if f.file? && !(formula_dir/"#{f.basename}.rb").exist?
  onoe "Broken alias: #{f}"
  true
end
```